### PR TITLE
chore: audit controller and management controller permissions

### PR DESCRIPTION
**Pull request by krancour**

### CODE REVIEW

File: charts/kargo/templates/controller/cluster-roles.yaml [hunk 1]
```diff
@@ -50,7 +50,6 @@ rules:
   - list
   - patch
   - promote
-  - update
```
Comment (krancour) [inline charts/kargo/templates/controller/cluster-roles.yaml:53]: The only update to Stages involves removing a finalizer and that can quite easily be done with a patch.

File: charts/kargo/templates/controller/cluster-roles.yaml [hunk 2]
```diff
@@ -40,7 +40,6 @@ rules:
   - get
   - list
   - watch
-  - patch
```
Comment (krancour) [inline charts/kargo/templates/controller/cluster-roles.yaml:43]: The controller doesn't need to patch Freight or Promotions at all.

File: charts/kargo/templates/controller/cluster-roles.yaml [hunk 3]
```diff
@@ -57,7 +57,6 @@ rules:
   verbs:
   - get
   - list
-  - patch
```
Comment (krancour) [inline charts/kargo/templates/controller/cluster-roles.yaml:62]: The controller does not need to patch Warehouses at all.

File: charts/kargo/templates/controller/cluster-roles.yaml [hunk 4]
```diff
@@ -67,7 +67,6 @@ rules:
   - warehouses/finalizers
   - warehouses/status
   verbs:
-  - update
```
Comment (krancour) [inline charts/kargo/templates/controller/cluster-roles.yaml:73]: The controller only ever patches status subresources.

File: charts/kargo/templates/controller/cluster-roles.yaml [hunk 5]
```diff
@@ -114,7 +114,6 @@ rules:
   - create
   - get
   - list
-  - patch
```
Comment (krancour) [inline charts/kargo/templates/controller/cluster-roles.yaml:121]: I don't believe there is actually any need to patch AnalysisRuns. Everything works correctly e2e without this.

@hiddeco maybe you can give me a sanity check on this one, however.

File: charts/kargo/templates/controller/cluster-roles.yaml [hunk 6]
```diff
@@ -106,6 +98,14 @@ metadata:
...
     {{- include "kargo.controller.labels" . | nindent 4 }}
 rules:
+- apiGroups:
+  - argoproj.io
+  resources:
+  - analysistemplates
+  verbs:
+  - get
+  - list
+  - watch
```
Comment (krancour) [inline charts/kargo/templates/controller/cluster-roles.yaml:108]: These permissions are better off down here where they will not be granted in the event that Rollouts integration is completely disabled.

File: charts/kargo/templates/management-controller/cluster-roles.yaml [hunk 7]
```diff
@@ -16,7 +16,6 @@ rules:
   - get
   - list
   - patch
-  - update
```
Comment (krancour) [inline charts/kargo/templates/management-controller/cluster-roles.yaml:19]: Not used at all.

File: charts/kargo/templates/management-controller/cluster-roles.yaml [hunk 8]
```diff
@@ -46,5 +46,4 @@ rules:
   - projects/status
   verbs:
   - patch
-  - update
```
Comment (krancour) [inline charts/kargo/templates/management-controller/cluster-roles.yaml:50]: This permission was already unused.

File: api/v1alpha1/helpers.go [hunk 9]
```diff
@@ -33,14 +33,6 @@ func AddFinalizer(ctx context.Context, c client.Client, obj client.Object) error
 	return nil
 }
 
-func ClearAnnotations(ctx context.Context, c client.Client, obj client.Object, keys ...string) error {
-	kvs := make(map[string]*string, len(keys))
-	for _, k := range keys {
-		kvs[k] = nil
-	}
-	return patchAnnotations(ctx, c, obj, kvs)
-}
```
Comment (krancour) [inline api/v1alpha1/helpers.go]: Seems we stopped clearing refresh annotation quite some time ago and that was the only thing we were ever using this for.

File: charts/kargo/templates/controller/cluster-roles.yaml (hunk 5 above)
Comment (krancour) [reply charts/kargo/templates/controller/cluster-roles.yaml:121]: nm. I just noticed the DM you sent me about this hours ago.

It's for aborting a verification's underlying AnalysisRun.

I'll fix this in the morning.

**Review by hiddeco — APPROVED** [review]

**Pull request merged**